src/app/customer/CheckoutButtonList.tsx

```diff
@@ -33,7 +33,7 @@ const CheckoutButtonList: FunctionComponent<CheckoutButtonListProps> = ({
     methodIds,
     ...rest
 }) => {
-    const supportedMethodIds = methodIds
+    const supportedMethodIds = (methodIds || [])
```

Comment:
Might be better defined in the deconstruct:

```
methodIds = [],
```

or at least use the  nullish coalescing operator:

```suggestion
    const supportedMethodIds = (methodIds ?? [])
```

Reply:
I think the constructor wouldn't work if `null` is explicitly passed?

Reply:
True, I'm not sure `null` is allowed as a replacement of `undefined` though.

Reply:
that's true, but we usually define types as `supportedMethodIds?: string[]` which is the same as `supportedMethodIds: string[] | undefined`. However, this value comes from the API and it can actually be `null`. For simplicty, we haven't really explicitly modelled `null` in the types (e.g. `supportedMethodIds?: string[] | null`).

Reply:
That can be an issue though, if we think the API might return `null` we might be better off typing it explicitly, otherwise someone can make the wrong assumption.